### DISCUSSION TITLE
tamper-resistant / digested downloading inside dockerfiles w/ regex

### DISCUSSION THREAD

**Discussion by novas0x2a** [renovatebot/renovate#11007] [Suggest an Idea]
I'm having trouble putting some pieces together with the regex manager!

In our dockerfiles, we follow a convention that looks like:

```dockerfile
ENV MDBOOK_VERSION=v0.4.4
ENV MDBOOK_SHA256=a518d005ed6a264e49998917d33e08769ac1d9c812405e7d214673aea625411d
```

And then we check the hash when we actually download it. I'm trying to expand the "Advanced Capture" [regex example](https://docs.renovatebot.com/modules/manager/regex/) to support this, e.g.: 

```dockerfile
# renovate: datasource=github-releases depName=rust-lang/mdBook
ENV MDBOOK_VERSION=v0.4.4
ENV MDBOOK_SHA256=a518d005ed6a264e49998917d33e08769ac1d9c812405e7d214673aea625411d
```

but I realized that the digest isn't really available at all from the github API, and thus, obviously, renovate can't really provide it either. (We've been generating the digest by hand for this dep).

I'm curious what conventions others might be following to handle this and similar situations:
1) No digest provided at all ([e.g.](https://github.com/rust-lang/mdBook/releases/tag/v0.4.11)). Is there somewhere else anyone knows of that's generating digests? (I'm thinking of something like repology, for example, something that exists as kind of a tracker overlay).
2) Digest provided in a separate file ([e.g.](https://github.com/open-policy-agent/conftest/releases/tag/v0.26.0)). Is it possible to handle these today with renovate? I don't see anything in the github-releases datasource that handles this, but I'm curious if there's a different way to handle it.

As an example of "a different way": we have a hack for any tools that release docker images (using the docker datasource instead), which is basically:
```dockerfile
FROM upstream/thing:v1.2.3@sha256:e641af12c52636cd19f5ca0c98acebdb73bd09ce2df52247e4cf2c93a4049346 as thing-download
FROM base as release
COPY --from=thing-download /usr/local/bin/thing /usr/bin/
```

In other words, we leverage docker's built-in digesting to handle it. Unfortunately, this only works for a small % of deps. I'm curious if anyone has any similar alternate routes to get some sort of tamper-resistance :) I'm not particularly tied to our current convention, so if someone's doing this by following a different convention, I'd be glad to know!

**Comment by viceice** [comment]
Checkout #10947

It will probably provide the missing hashes

**Reply by novas0x2a** [reply]
it kinda looks like someone read my mind, thanks! :)

**Reply by novas0x2a** [reply]
Especially when combined with https://github.com/renovatebot/renovate/issues/10993 this will solve a lot of problems for me!

**Reply by viceice** [reply]
#10993 won't work with #10947, as the `github-releases` datasource would need a manual hash, otherwise it couldn't know which hash of artifacts it the right one.
